internal/store/queuestore.go

```diff
@@ -97,6 +100,56 @@ func (store *QueueStore[_]) Open() error {
 	return nil
 }
 
+// Delete - Remove the store directory from disk
+func (store *QueueStore[_]) Delete() error {
+	return os.Remove(store.directory)
+}
+
+// PutMultiple - puts an item to the store.
+func (store *QueueStore[I]) PutMultiple(item []I) error {
+	store.Lock()
+	defer store.Unlock()
+	if uint64(len(store.entries)) >= store.entryLimit {
```

Comment:
We can try to add items until we reach the store limit. Let's say we have room for 5 more entries and the queue store receives a `PutMultiple` call with 10 items, we would not write even a single item, while we could have written 5. 

We could address this in a subsequent PR. I recognize that the store API can get proportionally complex if we were to address this.

Reply:
It's actually the reverse.. since items is a []item and the check if only comparing the total items already in the store against the limit, we would actually write a bit over the limit.

Reply:
Sorry, I misread that.